websites/B/Battlefield Portal/presence.ts

```diff
@@ -0,0 +1,121 @@
+const presence = new Presence({
+  clientId: "919182644296683520"
+});
+interface PlaygoundInfo {
+  playgroundId?: string;
+  type?: string;
+  playgroundName?: string;
+  playgroundDescription?: string;
+}
+
+async function getPlaygroundInfo(
+  playgroundId: string
+): Promise<string> {
+  const resp = await fetch(`https://api.gametools.network/bf2042/playground/?playgroundid=${playgroundId}`);
+  if (!resp.ok) throw new Error("Failed");
+  return resp.text();
+}
+
+let called = false,
+  playgroundId: string = null;
+
+const editingStamp = Math.floor(Date.now() / 1000),
+ presenceData: PresenceData = {
+    largeImageKey: "img1"
+  },
+  info : PlaygoundInfo = {};
+
+
+presence.on("UpdateData", async () => {
+  const [, sessionId] = document.cookie.match(
+      new RegExp("(^| )sessionId=([^;]+)")
+    ),
+    url = document.URL;
+
+  if (url.search("=") > 0) playgroundId = url.split("=").pop();
+
+  if (sessionId && playgroundId) {
+    info.playgroundId = playgroundId;
+    if (document.readyState === "complete") {
+      if (!called) {
+        called = true;
+        await GetPlaygroundInfo(playgroundId).then(t => {
+          const json = JSON.parse(t);
+          if (!json) {
+            called = false;
+            presenceData.details = null;
+          }else {
+            info.type = json.naming.type;
+            info.playgroundDescription = json.naming.playgroundDescription;
+            info.playgroundName = json.naming.playgroundName;
+            presence.info(JSON.stringify(info));
+          }
+        });
+      }
+    } else called = false;
+  }
+  if (info) {
+    presence
+      .getSetting("name")
+      .then((t) =>
+        t
+          ? (presenceData.details = `Making ${info.playgroundName}`)
+          : (presenceData.details = "")
```

Comment:
async/await should be used instead, and you shouldn't be setting details or state to an empty string. This notation is also not standard around here, same for all the other .then's

Reply:
Using Then is also global standard,  it's more readable imo, as the changes in it are only applicable after the async call is made

Reply:
Yeah but the assignment with parenthesis is not efficient, hence why I suggested using async/await and only setting the value if it exists, not set it to an empty string otherwise